luigi/contrib/hive.py

```diff
@@ -342,15 +347,19 @@ def run_job(self, job, tracking_url_callback=None):
                 query = query.encode('utf8')
             f.write(query)
             f.flush()
-            arglist = [load_hive_cmd(), '-f', f.name]
+            username_cmd = ""
+            if self.username:
+                username_cmd = "HADOOP_USER_NAME=" + self.username
+            arglist = [username_cmd] + [load_hive_cmd(), '-f', f.name]
```

Comment:
Does this really work? I always thought the environment variable setting in `MY_VAR=my_val echo hi` was done by the shell.